src/lpython/parser/tokenizer.re

```diff
@@ -89,13 +89,17 @@ void lex_int(Allocator &al, const unsigned char *s,
         s = s + 2;
         uint64_t n = get_value((char*)s, 2, loc);
         u.from_smallint(n);
-    } else if ((std::tolower(s[1]) == 'o')) {
+    } else if (std::tolower(s[1]) == 'o') {
         // Oct
         s = s + 2;
         uint64_t n = get_value((char*)s, 8, loc);
         u.from_smallint(n);
     } else {
         lex_dec_int_large(al, s, e, u);
+        if (s[0] == '0' && u.n != 0) {
```

Comment:
Hello @hankluo6 ! Thanks for this!

I think this is not the right file to make the changes. Your idea is correct. Please make the same changes in `tokenizer.cpp`. :+1: 

Welcome to LPython!

Reply:
Hi @kmr-srbh 

Thanks for your feedback! I think `tokenizer.cpp` is generated from `tokenizer.re` using `re2c` so I made change on this file. Please let me know if I'm mistaken.